garage/np/algos/base.py

```diff
@@ -1,7 +1,95 @@
-class Algorithm:
-    pass
+from collections import deque
 
+import numpy as np
 
-class RLAlgorithm(Algorithm):
-    def train(self):
-        raise NotImplementedError
+from garage.logger import tabular
+from garage.misc import special
+from garage.tf.misc import tensor_utils
+
+
+class RLAlgorithm():
```

Comment:
`class RLAlgorithm:`?

Reply:
I discussed with KR yesterday about it and this is the way he indicated. The `cem` and `cam_es` algorithms inherit from it instead of `np.BatchPolopt`. Actually `np.BatchPolopt` is pretty much not be used by anyone except some existing tests. By looking up the test for `cem` and `cam_es`, I found that it actually use our `OnPolicyVectorizedSampler`, which is inside tf. And also our `LocalRunner` is now a tf runner. I put the process_samples() here (`np.algos.RLAlgorithm`) for `cem` and `cam_es` use only, and rewrote some code to let them share this method without unnecessary duplicate implementation, as `cem` and `cam_es` use the same data to train.

To me it's actually more comfortable to make `RLAlgorithm` abstract and put the process_samples() for `cem` and `cam_es` into `np.BatchPolopt` and let `cem` and `cam_es` inherit from it. But that means I will need to remove some tests, such like `tests.fixtures.algos.InstrumentedBatchPolopt`. So my concern really is, am I allowed to do so?

I'm pretty new to the policy. So please correct me if I misunderstood something.